File: lib/src/main/java/com/orientsec/easysocket/session/Reader.java
package com.orientsec.easysocket.session;

import com.orientsec.easysocket.HeadParser;

import java.io.IOException;

/**
 * Represents a reader interface for processing incoming messages.
 * This interface defines a method for reading and decoding messages.
 */
public interface Reader {
    /**
     * Reads and processes incoming messages.
     * This method is responsible for handling the decoding of message headers and packets.
     *
     * @throws IOException If an I/O error occurs during the reading process.
     * @throws Exception   If an error occurs during the decoding process, specifically
     *                     in {@link HeadParser#decodePacket(HeadParser.Head, byte[])}.
     */
    void read() throws Exception;
}

File: lib/src/commonMain/kotlin/com/orientsec/easysocket/session/Reader.kt
package com.orientsec.easysocket.session

/**
 * Represents a reader interface for processing incoming messages.
 * This interface defines a method for reading and decoding messages.
 */
interface Reader {
    fun start()

    /**
     * Reads and processes incoming messages.
     * This method is responsible for handling the decoding of message headers and packets.
     *
     * @throws Exception   If an error occurs during the reading or decoding process.
     */
    @Throws(Exception::class)
    suspend fun read()

    fun shutdown()
}
